apps/webapp/src/state/receive.ts

```diff
@@ -0,0 +1,60 @@
+import { Account } from '@penumbra-zone/types';
+import { AllSlices, SliceCreator } from '.';
+import { getAddressByIndex, getEphemeralAddress } from '../fetchers/address';
+
+export interface ReceiveSlice {
+  index: number;
+  ephemeral: boolean;
+  selectedAccount: Account | undefined;
+  previous: () => void;
+  next: () => void;
+  setIndex: (index: number) => void;
+  setEphemeral: (ephemeral: boolean) => void;
+  setSelectedAccount: () => Promise<void>;
+}
+
+export const createReceiveSlice = (): SliceCreator<ReceiveSlice> => (set, get) => {
```

Comment:
Isn't this slice completely redundant to the accounts slice? Why are we repeating this code here and not just using the accounts slice?

Reply:
Think we should consider combining these

Reply:
it`s not clear for me!more details possible

Reply:
`apps/extension/src/state/accounts.ts`
`apps/webapp/src/state/receive.ts`

These are basically duplicates of each other. Why not build a `AccountsSelectorComponent`, which has state management built in locally, so we don't need duplicates here?

Reply:
without zustand? just use useState and useMemo? we can't use zustand in ui package

Reply:
Sure, why not?

Reply:
`setSelectedAccount` has the same logic, but uses different methods to get the address (`getAddressByIndex, getEphemeralAddress` functions): webapp - from the extension, extension - wasm library

Reply:
I think it makes sense to move everything except selectedAccount to the local state of AccountsSelectorComponent

Reply:
Ok, sounds good to me